api/arceos_api/src/imp/mem.rs

```diff
@@ -4,13 +4,13 @@ cfg_alloc! {
 
     pub fn ax_alloc(layout: Layout) -> Option<NonNull<u8>> {
         if let Ok(vaddr) = axalloc::global_allocator().alloc(layout) {
-            Some(unsafe { NonNull::new_unchecked(vaddr.get() as _) })
+            Some(vaddr)
```

Comment:
These lines can be simplified to `axalloc::global_allocator().alloc(layout).ok()`